perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -2394,7 +2394,16 @@ public void setAttributeInNestedTransaction(PerunSession sess, UserExtSource use
 
 	@Override
 	public AttributeDefinition createAttribute(PerunSession sess, AttributeDefinition attribute) throws AttributeDefinitionExistsException {
-		return createAttribute(sess, attribute, true);
+		AttributeDefinition attributeToReturn = createAttribute(sess, attribute, true);
+
+		// try to initialize and register module
+		AttributesModuleImplApi module = getAttributesManagerImpl().getUninitiatedAttributesModule(sess, attributeToReturn);
+		if (module != null) {
+			getAttributesManagerImpl().initAttributeModule(module);
+			getAttributesManagerImpl().registerAttributeModule(module);
+		}
+
+		return attributeToReturn;
 	}
```

Comment:
We should have also opposite logic, which unregister module when attribute definition is deleted, otherwise, we could have messages about eg. virtual attribute changes even when such attribute doesn't exist anymore (at least until restart). We already do the same for attribute dependencies.

Reply:
Added.